internal/expr/parser.go

```diff
@@ -486,24 +521,76 @@ func (p *Parser) parseIdentifier() (string, bool) {
 	return "", false
 }
 
-// parseColumn parses a column made up of name bytes, optionally dot-prefixed by
+// skipEnclosedParentheses starts from a opening parenthesis '(' and skips until
+// its closing ')', taking into account comments and string literals in between.
+func (p *Parser) skipEnclosedParentheses() (bool, error) {
+	initialCol, initialLine := p.colNum(), p.lineNum
+
+	if !p.skipByte('(') {
+		return false, nil
+	}
+
+	parenCount := 1
+	for parenCount > 0 && p.pos != len(p.input) {
+		if ok, err := p.skipStringLiteral(); err != nil {
+			return false, err
+		} else if ok {
+			continue
+		}
+		if ok := p.skipComment(); ok {
+			continue
+		}
+
+		if p.skipByte('(') {
+			parenCount++
+			continue
+		} else if p.skipByte(')') {
+			parenCount--
+			continue
+		}
+		p.advanceByte()
+	}
+
+	if parenCount > 0 {
```

Comment:
We restore the state with a checkpoint before returning an error in `skipStringLiteral`. We should do it here as well to keep it consistent. It can be useful for error messages in function calling this one.

Reply:
Done, both `cp.restore` added when returning.